tokio-stream/tests/watch.rs

```diff
@@ -27,3 +27,34 @@ async fn message_not_twice() {
     drop(tx);
     task.await.unwrap();
 }
+
+#[tokio::test]
+async fn watch_stream_from_rx() {
+    let (tx, rx) = watch::channel("hello");
+
+    let mut stream = WatchStream::from(rx);
+
+    assert_eq!(stream.next().await.unwrap(), "hello");
+
+    tx.send("bye").unwrap();
+
+    assert_eq!(stream.next().await.unwrap(), "bye");
+}
+
+#[tokio::test]
+async fn watch_stream_new_on_change() {
+    let (tx, rx) = watch::channel("hello");
+
+    let mut stream = WatchStream::new_on_changed(rx);
+
+    let task = tokio::spawn(async move {
+        let value = stream.next().await.unwrap();
+        if value != "bye" {
+            panic!("received unexpected value: {:?}", value);
+        }
+    });
+
+    tx.send("bye").unwrap();
+
+    task.await.unwrap();
+}
```

Comment:
Since tests run on the current-thread runtime, this test is actually _guaranteed_ to call `send(bye)` before it calls `stream.next()`. This means that the test would also pass when using `new`.

Please use our tokio-test utilities to instead assert that `next` returns pending before the send, and then assert that it returns `"bye"` after the send.

Reply:
should be resolved by some rework of this test